docs/readme.txt

```diff
@@ -24,53 +24,52 @@ download from:
+- Initial support for macOS 10.14 and its dark mode.
+- Support for non-integer font sizes and arbitrary font weights.
+- New wxLZMA{Input,Output}Stream classes.
+- Add wxDataViewToggleRenderer::ShowAsRadio(), wxDisplay::GetPPI(),
+  wxGrid::SetCornerLabelValue(), wxHtmlEasyPrinting::SetPromptMode(),
+  wxJoystickEvent::GetButtonOrdinal(), wxToolbook::EnablePage().
 
 Some of the other improvements:
 
-- wxWidgets can now be built with CMake too.
-- Strings can now be translated differently depending on their context.
-- Converting between wxString and UTF-8 encoded std::string is now
-  simpler and unsafe wxString can now be disabled on the opt-in basis
-  (see http://wxwidgets.blogspot.com/2017/02/safer-s.html)
-- It is possible to use any window (e.g. wxCheckBox) as wxStaticBox label now.
-- Many improvements to accessibility support under MSW.
-- wxGraphicsContext now supports pens with fractional widths.
-- Support for XDG file layout under Unix.
-- Many bug fixes to the behaviour (including TAB navigation) and appearances,
-  especially in wxGTK3 and wxOSX ports.
-- wxDataViewCtrl items and headers can be formatted using simple markup
-  and it is simpler to combine to put items with checkboxes into it. Many bugs
-  and inconsistencies between platforms in this control have been fixed too.
-- Several enhancements to wxStyledTextCtrl including better support for
-  custom lexers and auto-completion.
-- Many improvements to the (still experimental) wxQt port.
-
-Additionally, the latest versions of compilers (e.g. MSVS 2017) and
-operating systems (macOS 10.12) are now supported and all the third
-party libraries have been updated to their latest versions.
+- There were again many improvements to the (still experimental) wxQt port.
+- Fix several bugs related to focus handling and TAB navigation in wxGTK.
+- Make it possible to control pagination in wxHTML more precisely.
+- Fix several problems with high-DPI displays.
+- wxOSX now uses native NSImage/UIImage representation for wxBitmap.
+- Support strike-through font attribute in XRC and wxDataViewCtrl markup too.
+- Support more than 4 buttons in wxJoystick.
+- Add wxwidgets.props property sheet file for MSVS users.
 
 Please refer to the detailed change log for the full list of changes:
 
 https://raw.githubusercontent.com/wxWidgets/wxWidgets/v3.1.2/docs/changes.txt
 
 
+Notice that this release is almost completely compatible with 3.1.1 at the API
+level, so upgrading to it if you're already using wxWidgets 3 should be
+straightforward.
+
+
 Changes since 3.0
 -----------------
 
 Compared to the stable 3.0.x series, this version brings too many
-improvements and even more bug fixes to list them them all. Here is the
+improvements and even more bug fixes to list them all, but here is the
 maximally condensed summary:
 
+- Build system improvements: support for new compilers (MSVS 2017, g++ 8) and
+  OS versions as well as an entirely new new CMake build system.
```

Comment:
```suggestion
  OS versions as well as an entirely new CMake build system.
```

Reply:
Fixed (in both places), thanks for noticing!